### PR TITLE
{geo}[foss/2023a] MODFLOW v6.4.4

### PR DESCRIPTION
**Pull request by casparvl**
(created using `eb --new-pr`)

Depends on: 
- [x] #19686

### CODE REVIEW

**Comment by casparvl** [comment]
@boegelbot please test @ jsc-zen3

**Comment by casparvl** [comment]
@boegelbot please test @ generoso

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/a6bd8ed795de2654f2ea94be7a2ac547 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3160.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/1fd6d63c8d1f441a99f7242d0d4865c5 for a full test report.

File: easybuild/easyconfigs/m/MODFLOW/MODFLOW-6.4.4-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+configopts = '-Ddebug=false -Dparallel=true'
+
+runtest = 'meson test'
+testopts = '--verbose --no-rebuild'
+
+sanity_check_commands = ['mf6 --version']
+
+sanity_check_paths = {
+    'files': ['bin/mf6', 'bin/mf5to6', 'bin/zbud6', 'lib/libmf6.%s' % SHLIB_EXT],
+    'dirs': ['bin', 'lib'],
```
Comment (boegel) [inline easybuild/easyconfigs/m/MODFLOW/MODFLOW-6.4.4-foss-2023a.eb:43]: no need to check for non-empty `bin`/`lib` dir if you're already checking for specific binaries/libraries

```suggestion
    'dirs': [],

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3

**Comment by casparvl** [comment]
Test report by @casparvl
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
tcn1.local.snellius.surf.nl - Linux RHEL 8.6, x86_64, AMD EPYC 7H12 64-Core Processor, Python 3.6.8
See https://gist.github.com/casparvl/27b70bba188c3800adb108bc07cbe244 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @casparvl!

**Pull request merged**